src/pymor/operators/numpy.py

```diff
@@ -324,10 +325,17 @@ def apply_inverse(self, V, mu=None, initial_guess=None, least_squares=False,
                     raise InversionError(f'{str(type(e))}: {str(e)}') from e
                 R = R.T
             else:
-                try:
-                    R = solve(self.matrix, V.to_numpy().T).T
-                except np.linalg.LinAlgError as e:
-                    raise InversionError(f'{str(type(e))}: {str(e)}') from e
+                if not hasattr(self, '_lu_factor'):
+                    try:
+                        self._lu_factor = lu_factor(self.matrix)
+                    except np.linalg.LinAlgError as e:
+                        raise InversionError(f'{str(type(e))}: {str(e)}') from e
+                    gecon = get_lapack_funcs('gecon', self._lu_factor)
+                    rcond, _ = gecon(self._lu_factor[0], np.linalg.norm(self.matrix, ord=1), norm='1')
+                    if rcond < np.finfo(np.float64).eps:
+                        logger = getLogger('pymor.operators.numpy.NumpyMatrixOperator.apply_inverse')
+                        logger.warning(f'Ill-conditioned matrix (rcond={rcond:.6g}): result may not be accurate.')
```

Comment:
Something I've asked myself over time: we also have the `logger` attribute of `NumpyMatrixOperator`, so we could simply
```suggestion
                        self.logger.warning(f'Ill-conditioned matrix (rcond={rcond:.6g}) in apply_inverse: result may not be accurate.')
```
This would change the prefix to `NumpyMatrixOperator`, however. Any thoughts?

Reply:
Yes, methods of subclasses of `BasicObject` generally should use `self.logger`. So I'd be in favor of the suggested change.

Reply:
Of course... I blindly copied what was used above...